### PR TITLE
Add feedback dialog for ZHA device reconfiguration

### PR DESCRIPTION
**Pull request by dmulcahey**
## Proposed change

This PR adds support for visual feedback for ZHA device reconfiguration.

**REQUIRES** https://github.com/home-assistant/core/pull/48447

https://user-images.githubusercontent.com/1335687/112768568-96335d80-8fea-11eb-97fe-fe506c773c0c.mov

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by dmulcahey** [comment]
@bramkragten when you have a sec... I am struggling w/ something silly with this. The dialog works and shows for the first invocation. I have to refresh the page to make it work again after it is dismissed... What am I missing?

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  public async showDialog(
+    params: ZHAReconfigureDeviceDialogParams
+  ): Promise<void> {
+    this._params = params;
+    this._subscribe(params);
+  }
+
+  protected render(): TemplateResult {
+    return html`
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts:43]:
```suggestion
	if (!this._params) {
		return html``;
	}
    return html`
```

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+        if (paperTextArea) {
+          const textArea = (paperTextArea.inputElement as IronAutogrowTextareaElement)
+            .textarea;
+          textArea.scrollTop = textArea.scrollHeight;
+        }
+      }
+    }
+  }
+
+  private _close(): void {
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts:92]:
```suggestion
  public closeDialog(): void {
```

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+    this._unsubscribe();
+    this._formattedEvents = "";
+    this._params = undefined;
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts:95]:
```suggestion
    this._params = undefined;
    fireEvent(this, "dialog-closed", { dialog: this.localName });
```

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+              `
+            : ""}
+        </div>
+        <paper-textarea
+          readonly
+          max-rows="10"
+          class="log"
+          value="${this._formattedEvents}"
+        >
+        </paper-textarea>
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts:73]: Let's put these technical details inside an `expansion-panel`

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+    this._params = params;
+    this._subscribe(params);
+  }
+
+  protected render(): TemplateResult {
+    return html`
+      <ha-dialog
+        open
+        hideActions
+        @closing="${this._close}"
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts:47]:
```suggestion
        @closing="${this.closeDialog}"
```

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        >
+        </paper-textarea>
+      </ha-dialog>
+    `;
+  }
+
+  private _handleMessage(message: any): void {
+    if (message.type === LOG_OUTPUT) {
+      this._formattedEvents += message.log_entry.message + "\n";
+      if (this.shadowRoot) {
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts:81]: `this.shadowRoot` will always be defined

File: src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    if (!this.hass) {
+      return;
+    }
+    this._active = true;
+    const data: any = { type: "zha/devices/reconfigure" };
+    data.ieee = params.device.ieee;
+    this._subscribed = this.hass.connection.subscribeMessage(
+      (message) => this._handleMessage(message),
+      data
+    );
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/zha/dialog-zha-reconfigure-device.ts:119]: We should move this to `data/zha`

**Comment by dmulcahey** [comment]
@bramkragten backend PR merged

**Review by bramkragten — APPROVED** [review]

**Pull request merged**